packages/ai/src/ToolResult.tsx

```diff
@@ -30,16 +49,39 @@ function getBorderColor(isSuccess: boolean, toolName: string) {
   }
   switch (toolName) {
     case 'query':
-      return 'border-green-500';
+      return 'border-gray-500';
     case 'chart':
       return 'border-blue-500';
     case 'answer':
-      return 'border-gray-500';
+      return 'border-green-500';
     default:
       return 'border-gray-500';
   }
 }
 
+// Helper function to render the custom message based on its type
+function renderCustomMessage(
```

Comment:
Looks good! I am applying the same change in openassistant/core to support persistant and serialization.